packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -1103,9 +1100,22 @@ function build(
     assert(key.match(/^[-\w.+/]+$/), X`invalid vatstore key`);
   }
 
-  function setBuildRootObject(buildRootObject) {
-    assert(!didRoot);
-    didRoot = true;
+  async function startVat() {
+    assert(!didStartVat);
+    didStartVat = true;
+    // consume the first (unmetered) turn to ensure vat module code runs with metering
```

Comment:
add to the comment "unlike dispatch.deliver or dispatch.notify, we don't need to unserialize any data here, so there's no opportunity for GC-based meter divergence. But when #4381 adds `vatParameters` to this event, perform that unserialization during the unmetered turn.